core/src/main/java/io/kestra/plugin/core/execution/UnsetVariables.java

```diff
@@ -0,0 +1,89 @@
+package io.kestra.plugin.core.execution;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.ExecutionUpdatableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Allow to unset execution variables."
+)
+@Plugin(
+    examples = {
+        @Example(
+            full = true,
+            title = "Set then unset variables",
+            code = """
+                id: variables
+                namespace: company.team
+
+                variables:
+                  name: World
+
+                tasks:
+                  - id: set-vars
```

Comment:
```suggestion
                  - id: set_vars
```
best to avoid - in task ids as those need to be escaped later when referencing